sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -1091,6 +1114,38 @@ private[hive] object HiveClientImpl {
       stats = readHiveStats(properties))
   }
 
+  /**
+   * This is the same process copied from the method `getTable()`
+   * of [[org.apache.hadoop.hive.ql.metadata.Hive]] to do some extra fixes for non-views.
+   * Methods of extracting multiple [[HiveTable]] like `getRawTablesByName()`
+   * should invoke this before return.
+   */
+  def extraFixesForNonView(tTable: MetaStoreApiTable): MetaStoreApiTable = {
+    // For non-views, we need to do some extra fixes
+    if (!(HiveTableType.VIRTUAL_VIEW.toString == tTable.getTableType)) {
+      // Fix the non-printable chars
+      val parameters: JMap[String, String] = tTable.getSd.getParameters
+      val sf: String = parameters.get(serdeConstants.SERIALIZATION_FORMAT)
```

Comment:
https://github.com/apache/hive/blame/master/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java#L1342 

Do we need to check null here?